test/datatypes/uncertainty-test.ts

```diff
@@ -424,4 +432,37 @@ describe('Uncertainty', () => {
     new Uncertainty(2, 3).greaterThanOrEquals(new Uncertainty(1, 1)).should.be.true();
     new Uncertainty(2, null).greaterThanOrEquals(new Uncertainty(1, 1)).should.be.true();
   });
+
+  it('should maintain consistency with date/datetime equality', () => {
+    const d1 = new Date(2022, 1, 1);
+    const d2 = new DateTime(2022, 1, 1);
+    const u1 = new Uncertainty(new Date(2022, 1, 1));
+    const u2 = new Uncertainty(new DateTime(2022, 1, 1));
+
+    // Dates do not equal datetimes
+    d1.equals(d2).should.be.false();
+    d2.equals(d1).should.be.false();
+
+    // Dates do not equal uncertainties
+    d1.equals(u1).should.be.false();
```

Comment:
Do you think that maybe this _should_ work?  It's good that it works in the comparison equals method, but... should we fix it in `Date.equals` / `DateTime.equals` as well?  Or does that cause issues elsewhere?

Reply:
I agree 100%.  Equality should be consistent and commutative.  I understand the visitor pattern, but I don't quite see how it would be employed here (specifically for equality).  But I do agree that having a single implementation would be ideal. I've filed #279 to track this.

Reply:
Thanks for adding the issue.  Will add any other discussion there.